### ISSUE TITLE
Uberjarring, Plugins, and SNAPSHOTs

### ISSUE THREAD

**Issue by royalaid** [technomancy/leiningen#2123]
I have been trying to build a project with `lein uberjar`. When removing the SNAPSHOT tag to build the project lein complains about SNAPSHOTs being included in the build. 
My project.clj is

``` clojure
(defproject bitvore-addtags-test "0.1.0"
  :description "FIXME: write description"
  :url "http://example.com/FIXME"
  :license {:name "Eclipse Public License"
            :url "http://www.eclipse.org/legal/epl-v10.html"}
  :dependencies [[http-kit "2.1.18"]
                 [org.clojure/clojure "1.8.0"]
                 [org.clojure/core.async "0.2.374"]
                 [org.clojure/tools.cli "0.3.3"]
                 [ring/ring-codec "1.0.0"]
                 [org.clojure/data.json "0.2.6"]]
  :main ^:skip-aot bitvore-addtags-test.core
  :target-path "target/%s"
  :profiles {:uberjar {:aot :all}
             :dev {:plugins [[org.clojars.rhg135/lein-hg "0.0.1"]]}})
```

And below is my profiles.clj

``` clojure
{:user {:plugins [[cider/cider-nrepl "0.12.0-SNAPSHOT"]
                  [refactor-nrepl "2.3.0-SNAPSHOT"]
                  [lein-ancient "0.6.8" :exclusions [org.clojure/clojure]]
                  [lein-localrepo "0.5.3"]
                  [jonase/eastwood "0.2.3"]
                  [lein-kibit "0.1.2" :exclusions [[org.clojure/tools.cli]
                                                   [org.clojure/clojure]]]]}}
{:repl {:dependencies [[org.clojure/tools.nrepl "0.2.12"]
                       [alembic "0.3.2"]]}}
```

Running `lein deps :tree` gives me

``` clojure
 [cider/cider-nrepl "0.12.0-20160401.051952-13"]
   [org.tcrawley/dynapath "0.2.3" :exclusions [[org.clojure/clojure]]]
 [clojure-complete "0.2.4" :exclusions [[org.clojure/clojure]]]
 [http-kit "2.1.18"]
 [org.clojure/clojure "1.8.0"]
 [org.clojure/core.async "0.2.374"]
   [org.clojure/tools.analyzer.jvm "0.6.9"]
     [org.clojure/core.memoize "0.5.8"]
       [org.clojure/core.cache "0.6.4"]
         [org.clojure/data.priority-map "0.0.4"]
     [org.clojure/tools.analyzer "0.6.7"]
     [org.clojure/tools.reader "1.0.0-alpha1"]
     [org.ow2.asm/asm-all "4.2"]
 [org.clojure/data.json "0.2.6"]
 [org.clojure/tools.cli "0.3.3"]
 [org.clojure/tools.nrepl "0.2.12" :exclusions [[org.clojure/clojure]]]
 [refactor-nrepl "2.3.0-20160324.145421-1"]
 [ring/ring-codec "1.0.0"]
   [commons-codec "1.6"]
```

and `lein with-profile uberjar deps :tree`

``` clojure
 [http-kit "2.1.18"]
 [org.clojure/clojure "1.8.0"]
 [org.clojure/core.async "0.2.374"]
   [org.clojure/tools.analyzer.jvm "0.6.9"]
     [org.clojure/core.memoize "0.5.8"]
       [org.clojure/core.cache "0.6.4"]
         [org.clojure/data.priority-map "0.0.4"]
     [org.clojure/tools.analyzer "0.6.7"]
     [org.clojure/tools.reader "1.0.0-alpha1"]
     [org.ow2.asm/asm-all "4.2"]
 [org.clojure/data.json "0.2.6"]
 [org.clojure/tools.cli "0.3.3"]
 [ring/ring-codec "1.0.0"]
   [commons-codec "1.6"]
```

I am guessing this is a bug but figured I would file an issue and bring attention to it.

**Comment by hyPiRion** [comment]
I don't think this is a bug. You're using snapshot plugins, and I'm pretty sure that they are present if you do `lein with-profile +uberjar deps :plugin-tree`.

I think putting the snapshot dependencies into `:dev` or `:repl` would fix this.

**Comment by royalaid** [comment]
The only problem with that is I will have to define the emacs plugins in every project. What gets me is why the plugins are being included in an uberjar. The uberjar should just be the files required to run the project and not the build tooling correct?

**Comment by hyPiRion** [comment]
The plugins themselves are not included in the uberjar, but any plugin may be used to generate things that gets included in the uberjar – either through hooks, or through modification of `:prep-tasks`. See #1967, perhaps in particular [this comment](https://github.com/technomancy/leiningen/issues/1967#issuecomment-135758790).

**Comment by royalaid** [comment]
Thanks for pointing this out! That makes so much more sense. I have tried using dated versions of the plugins and still seems to trip the SNAPSHOT alert. An easy but annoying fix is to just remove nREPL and Refactor. Is there any built in way to override the profile merge? I have poked around on the wiki but didn't find anything leading me to believe so. Adding `{:plugins []}` in my `:uberjar` did not work.
